models/br_me_rais/br_me_rais__microdados_vinculos.sql

```diff
@@ -0,0 +1,100 @@
+    safe_cast(indicador_cei_vinculado as string) indicador_cei_vinculado,
+    safe_cast(indicador_trabalho_parcial as string) indicador_trabalho_parcial,
+    safe_cast(
+        indicador_trabalho_intermitente as string
+    ) indicador_trabalho_intermitente,
+    safe_cast(faixa_remuneracao_media_sm as string) faixa_remuneracao_media_sm,
+    safe_cast(valor_remuneracao_media_sm as float64) valor_remuneracao_media_sm,
+    safe_cast(valor_remuneracao_media as float64) valor_remuneracao_media,
+    safe_cast(faixa_remuneracao_dezembro_sm as string) faixa_remuneracao_dezembro_sm,
+    safe_cast(valor_remuneracao_dezembro_sm as float64) valor_remuneracao_dezembro_sm,
+    safe_cast(valor_remuneracao_janeiro as float64) valor_remuneracao_janeiro,
+    safe_cast(valor_remuneracao_fevereiro as float64) valor_remuneracao_fevereiro,
+    safe_cast(valor_remuneracao_marco as float64) valor_remuneracao_marco,
+    safe_cast(valor_remuneracao_abril as float64) valor_remuneracao_abril,
+    safe_cast(valor_remuneracao_maio as float64) valor_remuneracao_maio,
+    safe_cast(valor_remuneracao_junho as float64) valor_remuneracao_junho,
+    safe_cast(valor_remuneracao_julho as float64) valor_remuneracao_julho,
+    safe_cast(valor_remuneracao_agosto as float64) valor_remuneracao_agosto,
+    safe_cast(valor_remuneracao_setembro as float64) valor_remuneracao_setembro,
+    safe_cast(valor_remuneracao_outubro as float64) valor_remuneracao_outubro,
+    safe_cast(valor_remuneracao_novembro as float64) valor_remuneracao_novembro,
+    safe_cast(valor_remuneracao_dezembro as float64) valor_remuneracao_dezembro,
+    cast(
+        cast(regexp_replace(tipo_salario, r'^0+', '') as int64) as string
+    ) as tipo_salario,
+    safe_cast(valor_salario_contratual as float64) valor_salario_contratual,
+    safe_cast(subatividade_ibge as string) subatividade_ibge,
+    cast(
+        cast(regexp_replace(subsetor_ibge, r'^0+', '') as int64) as string
+    ) as subsetor_ibge,
+    safe_cast(cbo_1994 as string) cbo_1994,
+    safe_cast(cbo_2002 as string) cbo_2002,
+    safe_cast(cnae_1 as string) cnae_1,
+    safe_cast(cnae_2 as string) cnae_2,
+    safe_cast(cnae_2_subclasse as string) cnae_2_subclasse,
+    safe_cast(faixa_etaria as string) faixa_etaria,
+    safe_cast(idade as int64) idade,
+    safe_cast(grau_instrucao_1985_2005 as string) grau_instrucao_1985_2005,
+    safe_cast(grau_instrucao_apos_2005 as string) grau_instrucao_apos_2005,
+    safe_cast(nacionalidade as string) nacionalidade,
+    safe_cast(sexo as string) sexo,
+    safe_cast(raca_cor as string) raca_cor,
+    safe_cast(indicador_portador_deficiencia as string) indicador_portador_deficiencia,
+    safe_cast(tipo_deficiencia as string) tipo_deficiencia,
+    safe_cast(ano_chegada_brasil as int64) ano_chegada_brasil,
+    safe_cast(tamanho_estabelecimento as string) tamanho_estabelecimento,
+    case
+        when tipo_estabelecimento in ('Não', 'Não Indentificado')
```

Comment:
Não é melhor acrescentar esses tratamentos ao código de limpeza e rodar essa limpeza de novo? Assim fica meio arbitrário.

Reply:
Hmm, eu também sinto um incomodo de metade do processamento ser feito antes de suibr pro GCP e outra metade no dbt. A gente vem tentando reduzir ao máximo o tratamento fora do DBT, mas algumas tabelas é meio impossível de fazer isso pq precisa mexer em coisas antes de subir para big query pra funcionar,  então acabei aceitando que os tratamentos ficam metade metade.
Podemos manter tudo num lugar só, mas como essa tabela já teve todo seu tratamento feito antes do dbt ser uma coisa na BD tem alguns pontos: 
* Se for deixar tudo no código de limpeza a gente fica dependente de vc para qualquer modificação que for fazer.
* O correto no nosso novo sistema seria subir tudo de novo com o mínimo de tratamento possível e fazer todos os tratamentos no dbt, pq fica muito mais rápido pra gente fazer correções. 
* O principal problema aqui é que precisava fazer esses ajustes pra antes do lançamento da nova feature, pras conexões com dicionários não quebrarem, não sei se vai dar tempo

Reply:
Entendi! A RAIS é um bom caso para fazermos tudo via DBT mesmo então, porque a limpeza é relativamente simples. Uma outra coisa que me incomoda um pouco com esse caminho ELT é que parece menos transparente para quem quer auditar nosso código e só sabe Python ou R. Mas tudo bem, é um dos preços que pagamos.

Podemos fazer essa conversão para DBT puro depois do lançamento da feature então.